### PR TITLE
Allow OOM in Stress and Upgrade checks

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 7c49105cd56395bde38f0d7aff862c65c78c6989 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52807/7c49105cd56395bde38f0d7aff862c65c78c6989/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docker/test/stress/run.sh [hunk 1]
```diff
@@ -233,4 +233,10 @@ rowNumberInAllBlocks()
 LIMIT 1" < /test_output/test_results.tsv > /test_output/check_status.tsv || echo "failure\tCannot parse test_results.tsv" > /test_output/check_status.tsv
 [ -s /test_output/check_status.tsv ] || echo -e "success\tNo errors found" > /test_output/check_status.tsv
 
+# But OOMs in stress test are allowed
+if rg 'OOM in dmesg|Signal 9' /test_output/check_status.tsv
```
Comment (tavplubix) [inline docker/test/stress/run.sh:237]: "Signal 9" may also indicate that server got stuck during shoutdown and was killed

File: docker/test/stress/run.sh [hunk 2]
```diff
@@ -233,4 +233,10 @@ rowNumberInAllBlocks()
...
+then
+    sed -i 's/failure/success/' /test_output/check_status.tsv
```
Comment (tavplubix) [inline docker/test/stress/run.sh:239]: Should we keep "failure" status for debug builds? Memory tracking and memory consumption in debug builds should not be much different from release builds, so OOM with debug build may indicate a real problem

Comment (alexey-milovidov) [reply docker/test/stress/run.sh:239]: I've seen it frequently in debug builds. I'm not sure what the reason is - maybe the manual ASLR that we do.

**Comment by alexey-milovidov** [comment]
I've merged this PR to https://github.com/ClickHouse/ClickHouse/pull/52772 to make it ready, and now this PR is also merged, but we should refine it in a subsequent PR...

**Pull request merged**